### PR TITLE
[`GPTQ`, `CompressedTensors`] Fix unsafe imports and metada check

### LINKED ISSUE #34765: ModuleNotFoundError: No module named 'optimum' when loading model in TextGenerationWebUI
### System Info

- `transformers` version: 4.46.2
- Platform: macOS-15.1-arm64-arm-64bit-Mach-O
- Python version: 3.13.0
- Huggingface_hub version: 0.26.2
- Safetensors version: 0.4.5
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): not installed (NA)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I have cloned [text-generation-webui](https://github.com/oobabooga/text-generation-webui) on my M2 MacBook Pro and am trying to install a model from huggingface in it. I am following the steps and have chosen `Transformers` as the model loader, but am getting this error: 

```
23:48:52-670995 INFO     TRANSFORMERS_PARAMS=                                                                                    
{'low_cpu_mem_usage': True, 'torch_dtype': torch.float16}

23:48:52-672915 ERROR    Failed to load the model.                                                                               
Traceback (most recent call last):
  File "/Users/rob/IA/text-generation-webui/modules/ui_model_menu.py", line 232, in load_model_wrapper
    shared.model, shared.tokenizer = load_model(selected_model, loader)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/modules/models.py", line 93, in load_model
    output = load_func_map[loader](model_name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/modules/models.py", line 172, in huggingface_loader
    model = LoaderClass.from_pretrained(path_to_model, **params)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/installer_files/env/lib/python3.11/site-packages/transformers/models/auto/auto_factory.py", line 564, in from_pretrained
    return model_class.from_pretrained(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/installer_files/env/lib/python3.11/site-packages/transformers/modeling_utils.py", line 3652, in from_pretrained
    hf_quantizer = AutoHfQuantizer.from_config(config.quantization_config, pre_quantized=pre_quantized)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/installer_files/env/lib/python3.11/site-packages/transformers/quantizers/auto.py", line 148, in from_config
    return target_cls(quantization_config, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/IA/text-generation-webui/installer_files/env/lib/python3.11/site-packages/transformers/quantizers/quantizer_gptq.py", line 47, in __init__
    from optimum.gptq import GPTQQuantizer
ModuleNotFoundError: No module named 'optimum'

```

### Expected behavior

Model shoud load fine.

Thank you for your help. I'm pretty new with AI tools and any help is appreciated.

### PR DESCRIPTION
**Pull request by vasqu**
# What does this PR do?
Fixes some unsafe imports for GPTQ and CompressedTensors as well as an unsafe metadata check for GPTQ, i.e. metadata can't be checked if the package doesn't exist...

I tried looking into integrating the env validation function before even creating the quantizers but that would require a complete restructure, especially since the env is checked after creation to get additional info like device map etc. So I kept it as simple as I could and wrapped the unsafe stuff.

Minimal reproducible script to trigger gptq errors (when optimum is not installed):
```python
from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig


model_id = "facebook/opt-125m"
tokenizer = AutoTokenizer.from_pretrained(model_id)
dataset = ["auto-gptq is an easy-to-use model quantization library with user-friendly apis, based on GPTQ algorithm."]
gptq_config = GPTQConfig(bits=4, dataset=dataset, tokenizer=tokenizer)


quantized_model = AutoModelForCausalLM.from_pretrained(model_id, device_map="auto", quantization_config=gptq_config)
```

Fixes #34765

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@SunMarc @MekkCyber

### CODE REVIEW

File: src/transformers/quantizers/quantizer_gptq.py [hunk 1]
```diff
@@ -44,18 +44,29 @@ class GptqHfQuantizer(HfQuantizer):
...
+        if not is_optimum_available():
+            raise ImportError(
+                "Loading a GPTQ quantized model requires optimum (`pip install optimum`)"
+            )
+
+        if not is_auto_gptq_available():
+            raise ImportError(
+                "Loading a GPTQ quantized model requires the auto-gptq library (`pip install auto-gptq`)"
+            )
+
```
Comment (vasqu) [inline src/transformers/quantizers/quantizer_gptq.py:66]: Changed the order here so we can guarantee that the metadata check doesn't raise an error itself. Also separated but that's more of a preference of mine to differentiate, lmk if I should change this back to checking both at once.

Comment (MekkCyber) [reply src/transformers/quantizers/quantizer_gptq.py]: I think it's fine to have the checks separated, wdyt @SunMarc I think it's fine to have the checks separated, wdyt @SunMarc

**Comment by MekkCyber** [comment]
Thanks for the fix @vasqu, LGTM !

**Review by SunMarc — APPROVED** [review]
Thanks for the fixes !

File: src/transformers/quantizers/quantizer_gptq.py (hunk 1 above)
Comment (SunMarc) [reply src/transformers/quantizers/quantizer_gptq.py]: yes !

**Review by ArthurZucker — COMMENTED** [review]
Looks good but I think we should always call `validate_environment` before even initializing the quantizer (so probably update super (`HfQuantizer`) to call it at init time no?

**Comment by vasqu** [comment]
@ArthurZucker 
See my comments above: 
> I tried looking into integrating the env validation function before even creating the quantizers but that would require a complete restructure, especially since the env is checked after creation to get additional info like device map etc. So I kept it as simple as I could and wrapped the unsafe stuff.
The problem is some quantizers need additional information that is not available at creation; another option I could see it to add a post_init to create those unsafe parts :)

**Comment by vasqu** [comment]
gentle ping @ArthurZucker

**Review by ArthurZucker — APPROVED** [review]
Thanks for explaining

**Comment by ArthurZucker** [comment]
Not sure if it was fixed by recent quantizer compressed changes, let's resolve conflicts and good to merge

**Comment by vasqu** [comment]
Forgot about this PR myself :sweat_smile: should be good to merge now! Thx for the merges, ci sometimes be flaky tho

**Comment by MekkCyber** [comment]
will merge it once the ci is green @vasqu 🤗

**Pull request merged**